sys/include/delay.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2023 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @defgroup    sys_delay   Convenience functions to delay program execution
+ * @ingroup     sys
+ * @brief       Convenience functions to delay program execution
+ *
+ * # Motivation
+ *
+ * @ref sys_ztimer already provides @ref ztimer_sleep to allow delaying the
+ * execution of the running thread. While @ref ztimer_sleep allows via the
+ * selection of the clock to select the best fitting trade-off between
+ * resolution, accuracy (clock drift), power consumption and so on, it is
+ * difficult to write code with flexible requirements that just picks whatever
+ * clock is available.
```

Comment:
How about adding a paragraph on top here about what this module is supposed to do instead of having to derive it of what ztimer is not able to do. Then maybe include also a bit on when to pick this over ztimer.

Reply:
Done